src/ray/gcs/gcs_server/gcs_job_distribution.h

```diff
@@ -0,0 +1,357 @@
+// Copyright 2017 The Ray Authors.
```

Comment:
Everything in this file is only needed by the `GcsBasedActorScheduler`. I think we can just put them and `GcsBasedActorScheduler` in one file.